src/compile/common.ts

```diff
@@ -86,7 +86,29 @@ export function getMarkConfig<P extends keyof MarkConfig>(prop: P, mark: MarkDef
   return config.mark[prop];
 }
 
-export function formatSignalRef(fieldDef: FieldDef<string>, specifiedFormat: string, expr: 'datum' | 'parent', config: Config, useBinRange?: boolean) {
+export function formatSignalRef(fieldDef: FieldDef<string>, specifiedFormat: string, expr: 'datum' | 'parent', config: Config, formatType: FormatType | undefined, useBinRange?: boolean) {
+  // TODO: Refactor
```

Comment:
Can you just refactor rather than leaving it like this?

This seems to generate a very redundant chunk of code ...